### PR TITLE
Add support for error frames on PCAN

### PR DESCRIPTION
**Pull request by tamenol**
fix for #707

### CODE REVIEW

**Comment by tamenol** [comment]
What do I need to do to fix this? It's my first pull request ever on GitHub

**Comment by karlding** [comment]
Thanks! Looking at the failed CI job, it looks like you just need to run the [black](https://pypi.org/project/black/) code formatter.

Something like the following:
```bash 
# Install black formatter (or install all the lint requirements via requirements-lint.txt)
pip install black==19.3b0 
# Format project 
black .
```

**Comment by tamenol** [comment]
Thanks! Ran the black formatter and pushed the changes

**Comment by tamenol** [comment]
Do I need to add tests or something to the project?

File: can/interfaces/pcan/basic.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from string import *
 import platform
 import logging
+import winreg
```
Comment (karlding) [inline can/interfaces/pcan/basic.py]: Do you mind reverting these changes in `basic.py` to detect the registry key? Let's keep this PR's scope to adding support for error frames.

File: can/interfaces/pcan/pcan.py [hunk 2]
```diff
@@ -521,6 +547,14 @@ def _detect_available_configs():
             )
         return channels
 
+    def allow_error_frames(self, allow=True):
```
Comment (karlding) [inline can/interfaces/pcan/pcan.py]: I think we can just enable error frames in the `__init__` function for the `PcanBus`, much like the `SocketcanBus` does.

File: can/interfaces/pcan/basic.py (hunk 1 above)
Comment (tamenol) [reply can/interfaces/pcan/basic.py]: Ok will do. I make a new PR or it's not needed in the package?

File: can/interfaces/pcan/pcan.py (hunk 2 above)
Comment (tamenol) [reply can/interfaces/pcan/pcan.py]: Ok will do. Thanks

**Comment by tamenol** [comment]
Soo, I'd wait with this to pull into develop branch. Somehow it broke (or by doing it in the `__init__`  function or something I did outside of this package. But when initializing I get the `PCAN_ERROR_INITIALIZE' error. 
I'm reverting now to a stage I was in yesterday to see if it works again and than see what went wrong. My guess is that I shouldn't have put it before the call to `super()` in `__init__`.
I'm sorry for the inconvenience

**Comment by tamenol** [comment]
Ok, I found the culprit.

`result = self.m_objPCANBasic.SetValue(self.channel_info, PCAN_ALLOW_ERROR_FRAMES, PCAN_PARAMETER_ON)`
should be 
`result = self.m_objPCANBasic.SetValue(self.m_PcanHandle, PCAN_ALLOW_ERROR_FRAMES, PCAN_PARAMETER_ON)`

My bad.

I tested this, and it works. Somehow I copied old code over.

Testcode to do it yourself:
```
from can.interfaces.pcan.basic import *

channel_info = "PCAN_USBBUS1"
m_obj = PCANBasic()
chan = globals()[channel_info]

res = m_obj.Initialize(chan, PCAN_BAUD_500K, PCAN_TYPE_ISA, 11)
print(res)

res = m_obj.GetValue(chan, PCAN_ALLOW_ERROR_FRAMES)
print(res)

res = m_obj.SetValue(chan, PCAN_ALLOW_ERROR_FRAMES, PCAN_PARAMETER_ON)
print(res)
```

File: can/interfaces/pcan/pcan.py [hunk 3]
```diff
@@ -281,6 +288,32 @@ def status_is_ok(self):
         status = self.status()
         return status == PCAN_ERROR_OK
 
+    def status_string(self):
```
Comment (karlding) [inline can/interfaces/pcan/pcan.py]: Let's also revert this to keep this PR solely about the error frames

**Comment by karlding** [comment]
Thanks for working through this and catching your own bug! I have just one comment, otherwise it looks good.

**Comment by tamenol** [comment]
Done.

**Comment by tamenol** [comment]
Oh no. I'm sorry. Wanted to push this to my branch, but apperently it got merged in this PR. How to fix?

**Comment by karlding** [comment]
The easiest way (without rewriting history) is probably to do the following:

```bash
# Change directories to your git repo
cd python-can

# And then checkout the branch you're working on
git checkout ${BRANCH_NAME}

# Check commit history for the commit you want to revert to
# We will pick 5d7c5ec6438f1bab00cc262cca06b17f73e387cd
git log

# Note what files have changed between that commit and now.
# HEAD points to the top of your branch
git diff --stat 5d7c5ec6438f1bab00cc262cca06b17f73e387cd...HEAD
#  can/interfaces/pcan/basic.py | 10 +++++++++-
#  can/interfaces/pcan/pcan.py  | 25 +++++++++++++++++++++++++
#  2 files changed, 34 insertions(+), 1 deletion(-)

# Now restore each of those files at that commit
#
# In this case, since we want to restore all the files, we can
# just denote the current folder with .
git checkout 5d7c5ec6438f1bab00cc262cca06b17f73e387cd .

# Check the changes
git status
# On branch develop
# Your branch is up to date with 'origin/develop'.
# 
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
# 
#         modified:   can/interfaces/pcan/basic.py
#         modified:   can/interfaces/pcan/pcan.py

# Now create a new commit
git commit -m ${YOUR_COMMIT_MESSAGE}

# Push to origin
git push origin ${YOUR_BRANCH_NAME}
```

**Comment by tamenol** [comment]
Thank you so much for your patient and help. Fixed it.

**Comment by karlding** [comment]
Thanks for bringing this up and making a PR!

**Pull request merged**